### ISSUE TITLE
Crate Addition Request Apple Hypervisor Bindings

### ISSUE THREAD

**Issue by fabianpatras** [rust-vmm/community#129]
## Crate Name: apple-hv-bindings
Very much open to suggestions
## Short Description
There is this public repo: https://github.com/cloud-hypervisor/hypervisor-framework maintained by @sameo and @mxpv that adds wrappers around the [Apple Hypervisor API](https://developer.apple.com/documentation/hypervisor?language=objc).

This crate provides primitives needed for creating and managing VMs and vCPUs for VMM running on macOS.
## Why is this crate relevant to the rust-vmm project?
It would be a building block for creating vmms using [Apple Hypervisor API](https://developer.apple.com/documentation/hypervisor?language=objc) in Rust.

**Comment by sameo** [comment]
@fabianpatras Could you elaborate on what would this crate provide on top of the `hypervisor-framework` one?

**Comment by fabianpatras** [comment]
@sameo That is right, I'm only suggesting moving it from cloud-hypervisor org to the rust-vmm one. I just want to see your (cloud-hypervisor and rust-vmm people) opinions on this.

I intend to use [it](https://github.com/cloud-hypervisor/hypervisor-framework) for my diploma project ("rust-based vmm for macos") and I thought that it will be a great fit in rust-vmm organisation similarly to kvm and mshv bindings.

I can see that cloud-hypervisor org already [uses some crates from rust-vmm](https://github.com/cloud-hypervisor/cloud-hypervisor#4-rust-vmm-project-dependency) in its own repositories and I thought rust-vmm could really benefit from these Apple Hypervisor API Rust Bindings.
